server/src/main/kotlin/suwayomi/tachidesk/manga/impl/download/DownloadManager.kt

```diff
@@ -75,24 +80,75 @@ object DownloadManager {
         )
     }
 
-    suspend fun enqueue(chapterIndex: Int, mangaId: Int) {
-        if (downloadQueue.none { it.mangaId == mangaId && it.chapterIndex == chapterIndex }) {
-            downloadQueue.add(
-                DownloadChapter(
-                    chapterIndex,
-                    mangaId,
-                    chapter = ChapterTable.toDataClass(
-                        transaction {
-                            ChapterTable.select { (ChapterTable.manga eq mangaId) and (ChapterTable.sourceOrder eq chapterIndex) }
-                                .first()
+    @Serializable
+    data class EnqueueInput(
+        val mangaId: Int,
+        val chapterIndex: Int
+    )
+
+    fun enqueue(inputs: List<EnqueueInput>) {
+        val mangas = transaction {
+            val mangaIds = inputs.map { it.mangaId }.distinct()
+            MangaTable.select { MangaTable.id inList mangaIds }
+                .map { MangaTable.toDataClass(it) }
+        }
+
+        // This list will have unwanted chapters from other manga but there is no simple way
+        // to select only wanted ones. It will be mapped to input later so it should not be problem
+        val chapters = transaction {
+            val chapterIndexes = inputs.map { it.chapterIndex }.distinct()
+            ChapterTable.select { ChapterTable.sourceOrder inList chapterIndexes }.toList()
+        }
```

Comment:
I don't like this but I'll approve it since I can't think of a better way to do this

Reply:
I was thinking about this a bit more. Maybe we could just make it take chapter ID instead. Since this is a new API we could add id to chapters in all the endpoints and have this new one take chapter ID. All new batch API endpoints could then use chapter ids.

If this is added and then we would want to change it, it will make things a lot harder.